dali/test/python/test_pipeline_decorator.py

```diff
@@ -0,0 +1,89 @@
+# Copyright (c) 2021, NVIDIA CORPORATION. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from nvidia.dali.pipeline import Pipeline, pipeline
+from nose.tools import nottest
+import nvidia.dali.fn as fn
+from test_utils import get_dali_extra_path, compare_pipelines
+import os
+
+data_root = get_dali_extra_path()
+images_dir = os.path.join(data_root, 'db', 'single', 'jpeg')
+
+N_ITER = 7
+
+max_batch_size = 16
+num_threads = 4
+device_id = 0
+
+
+def reference_pipeline(flip_vertical, flip_horizontal, ref_batch_size=max_batch_size):
+    pipeline = Pipeline(ref_batch_size, num_threads, device_id)
+    with pipeline:
+        data, _ = fn.file_reader(file_root=images_dir)
+        img = fn.image_decoder(data, device="mixed")
+        flipped = fn.flip(img, horizontal=flip_horizontal, vertical=flip_vertical)
+        pipeline.set_outputs(flipped, img)
+    return pipeline
+
+
+@nottest
```

Comment:
why do you need nottest? The function won't be run automatically (not prefixed with test_). If you need that, why is it not there in `reference_pipeline`?

Reply:
To test, that pipeline decorator works with other decorators

Reply:
It's not obvious, maybe add a comment?

Reply:
Done